### ISSUE TITLE
Can load pre-train model from local disk

### ISSUE THREAD

**Issue by ZiyangYan** [LiheYoung/Depth-Anything#87]
I change the code as you shown in webpage, but it raised strange error, how can I fix it?
<img width="769" alt="1707484438184" src="https://github.com/LiheYoung/Depth-Anything/assets/92861079/9cc6aedd-5b1f-459c-aab7-e2eee882127b">
<img width="458" alt="1707484455187" src="https://github.com/LiheYoung/Depth-Anything/assets/92861079/15bc775d-64b2-4534-a129-102c1a9aac10">

And if I want to use metric_depth as pre-train model, can I also use the "from_pretrained" function to load it?

**Comment by pestrstr** [comment]
`DepthAnything.from_pretrained` with `local_files_only=True` tries to find a checkpoint on your local cache, previously downloaded from the web.

To load a local checkpoint, you can use the following code:
```
from depth_anything.dpt import DPT_DINOv2
depth_anything = DPT_DINOv2(encoder, features=64, out_channels=[48, 96, 192, 384])
ckpt = torch.load('models/checkpoints/depth_anything_vits14.pth')  
depth_anything.load_state_dict(ckpt)
```

this method works if you know hyper-params for the encoder that you want to load (`features` and `out_channels`).
To find the ones for the ViT-s encoder, I compared the model declaration in `depth_anything.dpt` with the model `state_dict`.
Maybe original authors can suggest us a better way to do it without explicitly specifying hyper-params.

**Comment by ZiyangYan** [comment]
> `DepthAnything.from_pretrained` with `local_files_only=True` tries to find a checkpoint on your local cache, previously downloaded from the web.
> 
> To load a local checkpoint, you can use the following code:
> 
> ```
> from depth_anything.dpt import DPT_DINOv2
> depth_anything = DPT_DINOv2('vits', features=64, out_channels=[48, 96, 192, 384])
> ckpt = torch.load('models/checkpoints/depth_anything_vits14.pth')  
> depth_anything.load_state_dict(ckpt)
> ```
> 
> this method works if you know hyper-params for the encoder that you want to load (`features` and `out_channels`). To find the ones for the ViT-s encoder, I compared the model declaration in `depth_anything.dpt` with the model `state_dict`. Maybe original authors can suggest us a better way to do it without explicitly specifying hyper-params.

Hi, thanks you for your reply, I changed my code follow your guidance, but it raised another error
![Uploading 1707490282887.png…]()

**Comment by ZiyangYan** [comment]
<img width="362" alt="1707490282887" src="https://github.com/LiheYoung/Depth-Anything/assets/92861079/875e2971-1196-4e88-87e7-27076233a80b">


> `DepthAnything.from_pretrained` with `local_files_only=True` tries to find a checkpoint on your local cache, previously downloaded from the web.
> 
> To load a local checkpoint, you can use the following code:
> 
> ```
> from depth_anything.dpt import DPT_DINOv2
> depth_anything = DPT_DINOv2('vits', features=64, out_channels=[48, 96, 192, 384])
> ckpt = torch.load('models/checkpoints/depth_anything_vits14.pth')  
> depth_anything.load_state_dict(ckpt)
> ```
> 
> this method works if you know hyper-params for the encoder that you want to load (`features` and `out_channels`). To find the ones for the ViT-s encoder, I compared the model declaration in `depth_anything.dpt` with the model `state_dict`. Maybe original authors can suggest us a better way to do it without explicitly specifying hyper-params.

**Comment by pestrstr** [comment]
Hi @ZiyangYan, did you move your model on GPU with `depth_anything.cuda()` after initializing it?
The error is reporting that your input is a CUDA tensor, while model weights are not

**Comment by ZiyangYan** [comment]
> Hi @ZiyangYan, did you move your model on GPU with `depth_anything.cuda()` after initializing it? The error is reporting that your input is a CUDA tensor, while model weights are not

Thanks, I can run it now, but do you know how to set metric model as pre-train model and run it?

**Comment by LiheYoung** [comment]
Hi @ZiyangYan, you can also follow the instructions [here](https://github.com/LiheYoung/Depth-Anything#no-network-connection-cannot-load-these-models). You need to download both the config file and checkpoint file. Then put them under the same directory, and specify this directory in ``DepthAnything.from_pretrained('the-directory-storing-config-and-ckp', local_files_only=True)``.

**Comment by ZiyangYan** [comment]
> Hi @ZiyangYan, to load our pre-trained models without Internet connection, you can suggested to follow the instructions [here](https://github.com/LiheYoung/Depth-Anything#no-network-connection-cannot-load-these-models). You need to download both the config file and checkpoint file. Then put them under the same directory, and specify this directory in `DepthAnything.from_pretrained('the-directory-storing-config-and-ckp', local_files_only=True)`.

Yes, I saw it and changed the code followed the instructions and then this problem occur, because the function search for the pre-train model in .cache directioy rather than the path set by me

**Comment by LiheYoung** [comment]
Hi @ZiyangYan, sorry for the late reply. If you have not solved this problem, please refer to our [latest instructions](https://github.com/LiheYoung/Depth-Anything#no-network-connection-cannot-load-these-models) for loading the models without Internet connection.

**Comment by frothmoon** [comment]
@ZiyangYan  Hello, how did you download the metric depth pre-training weight file? I've been having trouble connecting to huggingface.co. Could you send me a copy if it's convenient?

**Comment by heyoeyo** [comment]
The error: "Input type (torch.cuda.FloatTensor) and weight type (torch.FloatTensor) should be the same"

Is just saying that the input (the image) is on the gpu (cuda) whereas the model weights are not. To fix it, you'll just need to move the model to the gpu, which you can do by adding an extra line after you load the model:
```python
depth_anything = DepthAnything(model_configs[encoder])
depth_anything.load_state_dict(torch.load(f'./checkpoints/depth_anything_{encoder}14.pth'))
depth_anything.to(DEVICE)    # <---- This line moves the model to the gpu
```

**Comment by STRIVESS** [comment]
> The error: "Input type (torch.cuda.FloatTensor) and weight type (torch.FloatTensor) should be the same"
> 
> Is just saying that the input (the image) is on the gpu (cuda) whereas the model weights are not. To fix it, you'll just need to move the model to the gpu, which you can do by adding an extra line after you load the model:
> 
> ```python
> depth_anything = DepthAnything(model_configs[encoder])
> depth_anything.load_state_dict(torch.load(f'./checkpoints/depth_anything_{encoder}14.pth'))
> depth_anything.to(DEVICE)    # <---- This line moves the model to the gpu
> ```

Thank you so much! I run them successfully! 
![image](https://github.com/LiheYoung/Depth-Anything/assets/40514204/62bc7bf3-c0dd-447d-8433-ca2c9cc5a5b9)
![demo4_img_depth](https://github.com/LiheYoung/Depth-Anything/assets/40514204/57d7b263-7e51-4130-a8ad-66531836694c)
